internal/verifier/change_stream.go

```diff
@@ -98,20 +103,35 @@ func (verifier *Verifier) iterateChangeStream(ctx context.Context, cs *mongo.Cha
 		return err
 	}
 
-	readOneChangeEvent := func() (bool, error) {
-		gotEvent := cs.TryNext(ctx)
-		if gotEvent {
+	readAndHandleOneChangeEventBatch := func() (bool, error) {
+		eventsRead := 0
+		for hasEventInBatch := true; hasEventInBatch; hasEventInBatch = cs.RemainingBatchLength() > 0 {
```

Comment:
Checking RemainingBatchLength() before calling TryNext() seems like it risks missing events. I think there has to be a TryNext() before calling RemainingBatchLength() ?

Reply:
I think the for loop actually calls RemainingBatchLength after calling TryNext(). It's like 
```
for i := 0; i < 10; i++ {
	fmt.Print(i)
}
>> 0123456789
```